### PR TITLE
feat: move `models` and `webhooks` to `octocrate-types`

### LINKED ISSUE #4: Move `models` and `webhooks` to a seperates types crate
### Description

I'm using this code in a full-stack rust application which is being built using `leptos`. For this reason, I would like to share the same model structs between the frontend and backend.

Right now sharing the models and types also includes the `octocrate-core` crate which includes dependencies such as `jsonwebtoken` and others which would bloat the frontend build.

### PR DESCRIPTION
**Pull request by ifiokjr**
Fixes #4

### CODE REVIEW

File: types/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+security_advisories = []
+interactions = []
+classroom = []
+desktop = []
+
+[dependencies]
+serde_json = { workspace = true }
+serde = { workspace = true, features = ["derive"] }
+derive_builder = { workspace = true }
+chrono = { workspace = true, optional = true }
```
Comment (panghu-huang) [inline types/Cargo.toml:58]: `chrono` should be removable from `octocrate-types`

**Review by panghu-huang — CHANGES_REQUESTED** [review]
This can be temporarily implemented this way, but when codegen is regenerated, it will revert back. The best approach is to modify the directory where the code is generated in the codegen: https://github.com/panghu-huang/octocrate/blob/main/codegen/src/codegen.rs#L120

I can handle the codegen modification later on. Thank you for your contribution.

**Comment by ifiokjr** [comment]
Okay, I'm fixing.

**Comment by ifiokjr** [comment]
@panghu-huang this is ready to merge whenever you want.

**Review by panghu-huang — CHANGES_REQUESTED** [review]
https://github.com/panghu-huang/octocrate/blob/main/octocrate/src/installation_token.rs also needs to be moved to `octocrate-types`, now that it's split into two crates

BTW: I've added a simple CI check, which should take effect on the next commit!

File: codegen/src/main.rs [hunk 2]
```diff
@@ -13,11 +13,13 @@ fn main() {
   let codegen = Codegen::new();
 
   let start_time = Instant::now();
-  let parsed_api_descrption = codegen.parse();
+  let parsed_api_description = codegen.parse();
 
-  let dir = std::path::PathBuf::from("octocrate/src");
+  let apis_dir = std::path::PathBuf::from("octocrate/src");
+  let types_dir = std::path::PathBuf::from("octocrate-types/src");
```
[the remark this answers is not in the document]
Comment (ifiokjr) [reply codegen/src/main.rs:19]: Oh that's true thanks!

**Comment by panghu-huang** [comment]
Thank you

**Pull request merged**